Summary: Update emacs-git and emacs-nox11-git ports

Description:

This patch corresponds to a patch on pkgsrc main ports tree:
patch_emacs27-support.diff

* Update PLIST for emacs-git

* Update emacs-nox11-git mk-files such that the PLIST in emacs-git will
  be reused in the nox11 variant

* Update emacs-git mk-files to reuse a patched modules.mk in pkgsrc
  main -- a patch for supporting emacs-git and emacs-nox11-git as
  EMACS_TYPE emacs27 and emacs27nox respectively

* Add a subset of emacs-git options.mk options to a new emacs-nox11-git
  options.mk, supporting options gnutls and jansson in the nox11 variant

* Add version.mk for emacs-nox11-git, reusing emacs-git version.mk with
  modification per _EMACS_REQD - assuming an EMACS_TYPE updated per the 
  patch for pkgsrc main, denoted here

* Reuse the patched modules.mk from pkgsrc main in emacs-git,
  emacs-nox11-git

* Reuse the patched Makefile.common from pkgsrc main in emacs-git,
  emacs-nox11-git

* Reuse Makefile.common from emacs-git in emacs-nox11-git


# Additional Remarks

Emacs gnutls support would be required for users accessing ELPA package
repositories via HTTPS.

Cairo UI support is known to be usable in GTK+3 Emacs builds without rsvg
support i.e without pkgsrc build option svg. However, it may not be
certain as to whether Cairo UI support may contribute anything,
functionally -- directly, or via Pango, or otherwise -- to a build
without the 'svg' pkg option. In the series of local changesets addresed
with this patch, Cairo UI support had been tentatively added to all
GTK+3 Emacs builds, locally. That change was reverted before this patch
was produced.


# Changeset Information

Local author: Sean Champ <spchamp@users.noreply.github.com>

Changeset range: 907db8edb6 [upstream] to a0d7a1631e [local]
Changeset filter @ pkgsrc-wip: emacs*-git, local authorship

Upstream parent changeset @ pkgsrc-wip:              907db8edb6
Local parent changeset @ thinkum pkgsrc-wip contrib: d8f44f5f06

diff --git a/emacs-git/Makefile b/emacs-git/Makefile
index d1e03c0372..e6150e9fa2 100644
--- a/emacs-git/Makefile
+++ b/emacs-git/Makefile
@@ -1,65 +1,19 @@
-# $NetBSD: Makefile,v 1.74 2015/04/24 01:19:04 makoto Exp $
+# $NetBSD$
 
-.include	"version.mk"
-DISTNAME=	emacs-${_EMACS_VERSION_MAJOR}.${_EMACS_VERSION_MINOR}.${_EMACS_VERSION_MICRO}
-PKGNAME=	emacs27-${_EMACS_VERSION_MAJOR}.${_EMACS_VERSION_MINOR}.${_EMACS_VERSION_MICRO}
-CATEGORIES=	editors
-GIT_REPO=	git://git.savannah.gnu.org/emacs.git
+.include "Makefile.common"
+.include "version.mk"
 
-MAINTAINER=	pkgsrc-users@NetBSD.org
-HOMEPAGE=	https://www.gnu.org/software/emacs/
-COMMENT=	GNU editing macros (development version)
-LICENSE=	gnu-gpl-v3
-
-_GIT_FLAGS?=		--depth 1
+CONFLICTS=	emacs27-nox11-[0-9]*
 
 WRKSRC=		${WRKDIR}/emacs
 
-GNU_CONFIGURE=	yes
-INFO_FILES=	yes
-USE_TOOLS+=	autoconf automake gm4 gmake gsed gzip makeinfo
-# for example, Mac OS X 10.6.1 has 2.61 which fails
-AUTOCONF_REQD=	2.65
-# NetBSD has 4.8
-TEXINFO_REQD=	4.13
-
-CONFIGURE_ARGS+=	--srcdir=${WRKSRC}
-CONFIGURE_ARGS+=	--localstatedir=${VARBASE}
-CONFIGURE_ENV+=		GAMEOWN=${GAMES_USER}
-# For wip special, following gives symbol names when back trace is shown.
-CFLAGS+=		-g
-
-USE_GAMESGROUP=		yes
-
-BUILD_DEFS+=		VARBASE
-BUILD_DEFS+=		GAMEDATAMODE
-PLIST_SRC+=		PLIST
-
-MAKE_DIRS_PERMS+=	${VARBASE}/games/emacs ${GAMES_USER} ${GAMES_GROUP} ${GAMEDIRMODE}
-CONF_FILES_PERMS+=	/dev/null ${VARBASE}/games/emacs/snake-scores \
-			${GAMES_USER} ${GAMES_GROUP} ${GAMEDATAMODE}
-CONF_FILES_PERMS+=	/dev/null ${VARBASE}/games/emacs/tetris-scores \
-			${GAMES_USER} ${GAMES_GROUP} ${GAMEDATAMODE}
+PLIST_SRC=	${PKGDIR}/../emacs-git/PLIST
 
 .include "options.mk"
 
-# build PATH in the dumped emacs is not a problem
-CHECK_WRKREF_SKIP+=	bin/emacs
 CHECK_WRKREF_SKIP+=	bin/emacs-${PKGVERSION_NOREV}
 CHECK_WRKREF_SKIP+=	libexec/emacs/${PKGVERSION_NOREV}/${MACHINE_GNU_PLATFORM}/emacs.pdmp
 
-PRINT_PLIST_AWK+=	{ gsub(/${PKGVERSION_NOREV}/, "$${PKGVERSION}"); \
-			gsub(/${MACHINE_GNU_PLATFORM}/, "$${MACHINE_GNU_PLATFORM}"); \
-			print; next; }
-
-post-extract:
-	# set auto-load for send-pr library
-	cp ${FILESDIR}/site-init.el ${WRKSRC}/lisp
-
-pre-configure:
-	(cd ${WRKSRC}; ${PKGSRC_SETENV} ${CONFIGURE_ENV} ${CONFIG_SHELL} ./autogen.sh)
-
-.include "../../wip/mk/git-package.mk"
 .include "../../fonts/harfbuzz/buildlink3.mk"
 .include "../../graphics/hicolor-icon-theme/buildlink3.mk"
 .include "../../graphics/libimagequant/buildlink3.mk"
diff --git a/emacs-git/Makefile.common b/emacs-git/Makefile.common
new file mode 100644
index 0000000000..0ac64cd09b
--- /dev/null
+++ b/emacs-git/Makefile.common
@@ -0,0 +1,43 @@
+# $NetBSD$
+#
+# used by wip/emacs-git/Makefile
+# used by wip/emacs-nox11-git/Makefile
+
+.if !defined(EMACS_GIT_COMMON_MK)
+EMACS_GIT_COMMON_MK=	#defined
+
+.include "../../editors/emacs26/Makefile.common"
+
+PKGNAME=	emacs27-${_EMACS_VERSION_MAJOR}.${_EMACS_VERSION_MINOR}.${_EMACS_VERSION_MICRO}
+DISTNAME=	emacs-${_EMACS_VERSION_MAJOR}.${_EMACS_VERSION_MINOR}.${_EMACS_VERSION_MICRO}
+WRKSRC=		${WRKDIR}/emacs
+
+GIT_REPO=	git://git.savannah.gnu.org/emacs.git
+#_GIT_FLAGS?=	--depth 1
+## ^ default now. 
+## ^ To override, set GIT_DEEP_CLONE.default=yes local to this port
+
+# for example, Mac OS X 10.6.1 has 2.61 which fails
+AUTOCONF_REQD=	2.65
+# NetBSD has 4.8
+TEXINFO_REQD=	4.13
+
+## from an earlier wip/emacs-nox11-git/Makefile
+#REPLACE_PERL=		lib-src/grep-changelog
+#CHECK_INTERPRETER_SKIP=	bin/grep-changelog
+
+
+PRINT_PLIST_AWK+=	{ gsub(/${PKGVERSION_NOREV}/, "$${PKGVERSION}"); \
+			print; next; }
+
+post-extract:
+#	set auto-load for send-pr library
+	cp ${FILESDIR}/site-init.el ${WRKSRC}/lisp
+
+pre-configure:
+	(cd ${WRKSRC}; ${PKGSRC_SETENV} ${CONFIGURE_ENV} ${CONFIG_SHELL} ./autogen.sh)
+
+.include "../../wip/mk/git-package.mk"
+
+.endif
+
diff --git a/emacs-git/PLIST b/emacs-git/PLIST
index d1021b5af3..124bc113a0 100644
--- a/emacs-git/PLIST
+++ b/emacs-git/PLIST
@@ -6,7 +6,6 @@ bin/emacs-${PKGVERSION}
 bin/emacsclient
 bin/etags
 include/emacs-module.h
-info/ada-mode.info.gz
 info/auth.info.gz
 info/autotype.info.gz
 info/bovine.info.gz
@@ -3344,14 +3343,6 @@ share/emacs/${PKGVERSION}/lisp/proced.el.gz
 share/emacs/${PKGVERSION}/lisp/proced.elc
 share/emacs/${PKGVERSION}/lisp/profiler.el.gz
 share/emacs/${PKGVERSION}/lisp/profiler.elc
-share/emacs/${PKGVERSION}/lisp/progmodes/ada-mode.el.gz
-share/emacs/${PKGVERSION}/lisp/progmodes/ada-mode.elc
-share/emacs/${PKGVERSION}/lisp/progmodes/ada-prj.el.gz
-share/emacs/${PKGVERSION}/lisp/progmodes/ada-prj.elc
-share/emacs/${PKGVERSION}/lisp/progmodes/ada-stmt.el.gz
-share/emacs/${PKGVERSION}/lisp/progmodes/ada-stmt.elc
-share/emacs/${PKGVERSION}/lisp/progmodes/ada-xref.el.gz
-share/emacs/${PKGVERSION}/lisp/progmodes/ada-xref.elc
 share/emacs/${PKGVERSION}/lisp/progmodes/antlr-mode.el.gz
 share/emacs/${PKGVERSION}/lisp/progmodes/antlr-mode.elc
 share/emacs/${PKGVERSION}/lisp/progmodes/asm-mode.el.gz
@@ -3986,3 +3977,4 @@ share/icons/hicolor/scalable/apps/emacs.svg
 share/icons/hicolor/scalable/mimetypes/emacs-document.svg
 share/icons/hicolor/scalable/mimetypes/emacs-document23.svg
 share/metainfo/emacs.appdata.xml
+include/emacs-module.h
diff --git a/emacs-git/modules.mk b/emacs-git/modules.mk
index 2ac1eafb4e..f0abe0a7b4 100644
--- a/emacs-git/modules.mk
+++ b/emacs-git/modules.mk
@@ -1,353 +1,8 @@
-# $NetBSD: modules.mk,v 1.1 2013/10/22 13:01:48 makoto Exp $
-#
-# This Makefile fragment handles Emacs Lisp Packages (== ELPs).
-#
-# Note to users:
-#
-#	* Users choose one favourite Emacs version.
-#
-#	* Emacs Lisp files are installed...
-#		GNU emacs
-#			-> ${PREFIX}/share/emacs/site-lisp/foo/...
-#		XEmacs
-#			->${PREFIX}/lib/xemacs/site-packages/lisp/foo/...
-#
-#	* You can't install an ELP for both Emacs and XEmacs
-#	  simultaneously.
-#
-#	* XEmacs package's names are prefix'ed by "xemacs-".
-#
-# Note to package developers:
-#
-#	* XEmacs package's names are prefix'ed by "xemacs-".
-#
-#	  This is to make sure ELPs for XEmacs keep their own dependency
-#	  tree (for example, if an ELP "bar" depends on another ELP "foo",
-#	  "xemacs-bar" depends on "xemacs-foo".  "foo" (installed for GNU
-#	  Emacs) is useless for "xemacs-bar" in this case.).
-#
-#	  Make sure too that "foo" and "xemacs-foo" conflict each other,
-#	  since they can share some files.  (Making all ELPs separately
-#	  installable for GNU Emacs/XEmacs might be possible.)
-#
-#	* Assume each ELP supports all Emacs versions by default.  If the
-#	  ELP supports only certain Emacs versions, define EMACS_VERSIONS
-#	  _ACCEPTED explicitly before including mk/emacs.mk.
-#
-# Variables for users:
-#
-#	EMACS_TYPE
-#		Description:
-#			The user's favourite Emacs version.  The default
-#			value is set in mk/defaults/mk.conf.
-#		Possible values:
-#			emacs25, emacs25nox, emacs21, emacs21nox, emacs20,
-#			xemacs215, xemacs215nox, xemacs214, xemacs214nox
-#		Default value:
-#			emacs25
-#
-# Variables ELPs can provide:
-#
-#	EMACS_MODULES
-#		Description:
-#			List of Emacs LISP modules that the current ELP
-#			needs at run-time.
-#		Possible values:
-#			base, leim
-#		Default value:
-#			<undefined>
-#
-#	EMACS_VERSIONS_ACCEPTED
-#		Description:
-#			Versions the ELP accepts (supports).
-#		Possible values:
-#			emacs25, emacs25nox, emacs21, emacs21nox, emacs20,
-#			xemacs215, xemacs215nox, xemacs214, xemacs214nox
-#		Default value:
-#			emacs25, emacs25nox, emacs21, emacs21nox, emacs20,
-#			xemacs215, xemacs215nox, xemacs214, xemacs214nox
-#
-#	EMACS_BUILDLINK
-#		Description:
-#			Whether create buildlink directory of *.el / *.elc
-#			files.  ELPs which include other ELP's buildlink3.mk
-#			must define this value.
-#		Possible values:
-#			<defined>, <undefined>
-#		Default value:
-#			<undefined>
-#
-# Variables provided for ELPs:
-#
-#	EMACS_BIN
-#		Description:
-#			Path to Emacs executable.
-#		Possible values:
-#			${PREFIX}/bin/emacs, ${PREFIX}/bin/xemacs
-#
-#	EMACS_ETCPREFIX
-#		Description:
-#			Path to the directory misc. files should be
-#			installed into.  ELPs should append a short
-#			name as a subdirectory.
-#		Possible values:
-#			${PREFIX}/share
-#			${PREFIX}/lib/xemacs/site-packages/etc
-#
-#	EMACS_FLAVOR
-#		Description:
-#			GNU Emacs (emacs) or XEmacs (xemacs).
-#		Possible values:
-#			emacs, xemacs
-#
-#	EMACS_INFOPREFIX
-#		Description:
-#			Path to the directory Info files should be
-#			installed into.  Unlike EMACS_ETCPREFIX or
-#			EMACS_LISPPREFIX, subdirectory is not needed.
-#		Possible values:
-#			${PREFIX}/${PKGINFODIR}
-#			${PREFIX}/lib/xemacs/site-packages/info
-#
-#	EMACS_LISPPREFIX
-#		Description:
-#			installed into.  ELPs should append a short name
-#			as a subdirectory.
-#		Possible values:
-#			${PREFIX}/share/emacs/site-lisp
-#			${PREFIX}/lib/xemacs/site-packages/lisp
-#
-#	EMACS_PKGNAME_PREFIX
-#		Description:
-#			The prefix of PKGNAME and DEPENDS lines.  All ELPs
-#			must honour this!
-#		Possible values:
-#			"", "xemacs-"
-#
-#	EMACS_VERSION_MAJOR
-#		Description:
-#			Emacs major version.
-#		Possible values:
-#			20, 21, 22, 23, 24, 25, <integers more than that in the future>
-#
-#	EMACS_VERSION_MINOR
-#		Description:
-#			Emacs minor version.
-#		Possible values:
-#			<integer>
-#
-#	EMACS_VERSION_MICRO
-#		Description:
-#			Emacs micro version.
-#		Possible values:
-#			<integer>
-#
-# Variables provided in ELPs' PLIST:
-#
-#	EMACS_ETCPREFIX
-#		Description:
-#			Same as the one in Makefile, except that
-#			${PREFIX} is omitted in PLIST.
-#		Possible values:
-#			share
-#			lib/xemacs/site-packages/etc
-#
-#	EMACS_INFOPREFIX
-#		Description:
-#			Same as the one in Makefile, except that
-#			${PREFIX} is omitted in PLIST.
-#		Possible values:
-#			${PKGINFODIR}
-#			lib/xemacs/site-packages/info
-#
-#	EMACS_LISPPREFIX
-#		Description:
-#			Same as the one in Makefile, except that
-#			${PREFIX} is omitted in PLIST.
-#		Possible values:
-#			share/emacs/site-lisp
-#			lib/xemacs/site-packages/lisp
-#
-#	EMACS_VERSION
-#		Description:
-#			XXX
-#		Possible values:
-#			XXX
-#
-#	FOR_{emacs25,emacs21,emacs21nox,emacs20,xemacs215,xemacs215nox,xemacs214,xemacs214nox}
-#	FOR_{emacs,xemacs}
-#	FOR_{emacs_x,emacs_nox}
-#	NOTFOR_{emacs25,emacs21,emacs21nox,emacs20,xemacs215,xemacs215nox,xemacs214,xemacs214nox}
-#	NOTFOR_{emacs,xemacs}
-#	NOTFOR_{emacs_x,emacs_nox}
-#		Description:
-#			These macros will become either an empty string or
-#			"@comment" depending on the Emacs version; when
-#			"emacs21" is used, ${FOR_emacs} and ${FOR_emacs21}
-#			become "" (empty), and other macros become
-#			"@comment"; this means that only the PLIST lines
-#			prefixed by ${FOR_emacs} and ${FOR_emacs21} are
-#			valid.
-#
-#			NOTFOR_* is the opposite of FOR_*.
-#		Possible values:
-#			"", "@comment"
-#
-#	FOR_emacs_no_byte_compile
-#		Description:
-#			For .elc file of .el file marked as "no-byte-compile: t"
-#		Possible values:
-#			"", "@comment"
+# $NetBSD$
 
-.if !defined(EMACS_MK)
-EMACS_MK=	# defined
+.if !defined(EMACS_GIT_MODULES_MK)
+EMACS_GIT_MODULES_MK=	# defined
 
-_VARGROUPS+=		emacs
-_USER_VARS.emacs=	EMACS_TYPE
-_PKG_VARS.emacs=	EMACS_MODULES EMACS_VERSIONS_ACCEPTED
-_SYS_VARS.emacs=	EMACS_BIN EMACS_ETCPREFIX EMACS_FLAVOR \
-			EMACS_INFOPREFIX EMACS_LISPPREFIX \
-			EMACS_PKGNAME_PREFIX \
-			EMACS_VERSION_MAJOR EMACS_VERSION_MINOR \
-			EMACS_VERSION_MICRO
-_DEF_VARS.emacs=	_EMACS_TYPE _EMACS_PKGBASE
-BUILD_DEFS+=		${_USER_VARS.emacs}
-BUILD_DEFS_EFFECTS+=	${_SYS_VARS.emacs}
+.include "../../editors/emacs/modules.mk"
 
-.include "../../mk/bsd.prefs.mk"
-
-#
-# Constants
-#
-
-_EMACS_VERSIONS_ALL= \
-	emacs20 emacs21 emacs21nox emacs25 emacs26 emacs27\
-	xemacs214 xemacs214nox xemacs215 xemacs215nox
-
-_EMACS_PKGDIR_MAP= \
-	emacs20@../../editors/emacs20 \
-	emacs21@../../editors/emacs21 \
-	emacs21nox@../../editors/emacs21-nox11 \
-	emacs25@../../editors/emacs25 \
-	emacs25nox@../../editors/emacs25-nox11 \
-	emacs26@../../editors/emacs26 \
-	emacs27@../../wip/emacs-git \
-	xemacs214@../../editors/xemacs \
-	xemacs214nox@../../editors/xemacs-nox11 \
-	xemacs215@../../editors/xemacs-current \
-	xemacs215nox@../../editors/xemacs-current-nox11
-
-_EMACS_ETCDIR.emacs=		share
-_EMACS_INFODIR.emacs=		${PKGINFODIR}
-_EMACS_LISPDIR.emacs=		share/emacs/site-lisp
-_EMACS_PKGNAME_PREFIX.emacs=
-_EMACS_CONFLICTS.emacs=		xemacs-${PKGBASE}-[0-9]*
-
-_EMACS_ETCDIR.xemacs=		lib/xemacs/site-packages/etc
-_EMACS_INFODIR.xemacs=		lib/xemacs/site-packages/info
-_EMACS_LISPDIR.xemacs=		lib/xemacs/site-packages/lisp
-_EMACS_PKGNAME_PREFIX.xemacs=	xemacs-
-_EMACS_CONFLICTS.xemacs=	${PKGBASE:C|^xemacs-||}-[0-9]*
-
-#
-# Version decision
-#
-# 1. Pick up an emacs version.
-#
-#	add a dependency to the version;
-#
-# 2. Check if a given package can be used for the emacs version.
-#
-# 	if (package accepts the emacs version)
-#		nothing to do;
-#	else
-#		abort;
-#
-
-_EMACS_TYPE=	${EMACS_TYPE}
-
-EMACS_VERSIONS_ACCEPTED?=	${_EMACS_VERSIONS_ALL}
-.if empty(EMACS_VERSIONS_ACCEPTED:M${_EMACS_TYPE})
-PKG_FAIL_REASON+=		"Accepted versions are: ${EMACS_VERSIONS_ACCEPTED}"
-PKG_FAIL_REASON+=		"No valid Emacs version installed found"
-.endif
-
-_EMACS_PKGDIR=	${_EMACS_PKGDIR_MAP:M${_EMACS_TYPE}@*:C|${_EMACS_TYPE}@||}
-
-.include "${_EMACS_PKGDIR}/version.mk"
-
-#
-# Dependencies and conflicts
-#
-
-DEPENDS+=	${_EMACS_REQD}:${_EMACS_PKGDIR}
-CONFLICTS+=	${_EMACS_CONFLICTS.${_EMACS_FLAVOR}}
-
-EMACS_MODULES?=
-.for _mod_ in ${EMACS_MODULES}
-.  if !empty(_EMACS_PKGDEP.${_mod_})
-DEPENDS+=	${_EMACS_PKGDEP.${_mod_}}
-.  endif
-.endfor
-
-#
-# Macros for packages
-#
-
-EMACS_FLAVOR=		${_EMACS_FLAVOR}
-EMACS_BIN=		${PREFIX}/bin/${_EMACS_FLAVOR}
-EMACS_VERSION_MAJOR=	${_EMACS_VERSION_MAJOR}
-EMACS_VERSION_MINOR=	${_EMACS_VERSION_MINOR}
-EMACS_VERSION_MICRO=	${_EMACS_VERSION_MICRO}
-EMACS_ETCPREFIX=	${PREFIX}/${_EMACS_ETCDIR.${_EMACS_FLAVOR}}
-EMACS_INFOPREFIX=	${PREFIX}/${_EMACS_INFODIR.${_EMACS_FLAVOR}}
-EMACS_LISPPREFIX=	${PREFIX}/${_EMACS_LISPDIR.${_EMACS_FLAVOR}}
-EMACS_PKGNAME_PREFIX=	${_EMACS_PKGNAME_PREFIX.${_EMACS_FLAVOR}}
-
-GNU_CONFIGURE_INFODIR?=	${EMACS_INFOPREFIX}
-
-_EMACS_PLIST_SUBST=
-.for e in ${_EMACS_VERSIONS_ALL} emacs xemacs
-.  if "${e}" == ${EMACS_FLAVOR} || "${e}" == ${_EMACS_TYPE}
-_EMACS_PLIST_SUBST+=	FOR_${e}="" NOTFOR_${e}="@comment "
-.  else
-_EMACS_PLIST_SUBST+=	FOR_${e}="@comment " NOTFOR_${e}=""
-.  endif
-.endfor
-.if empty(EMACS_TYPE:M*nox)
-_EMACS_PLIST_SUBST+=	FOR_emacs_x="" NOTFOR_emacs_x="@comment "
-_EMACS_PLIST_SUBST+=	FOR_emacs_nox="@comment " NOTFOR_emacs_nox=""
-.else
-_EMACS_PLIST_SUBST+=	FOR_emacs_x="@comment " NOTFOR_emacs_x=""
-_EMACS_PLIST_SUBST+=	FOR_emacs_nox="" NOTFOR_emacs_nox="@comment "
-.endif
-
-_EMACS_PLIST_SUBST+=	EMACS_FLAVOR=${EMACS_FLAVOR:Q}
-_EMACS_PLIST_SUBST+=	EMACS_VERSION=${_EMACS_VERSION_MAJOR:Q}.${_EMACS_VERSION_MINOR:Q}
-_EMACS_PLIST_SUBST+=	EMACS_ETCPREFIX=${EMACS_ETCPREFIX:C|^${PREFIX}/||}
-_EMACS_PLIST_SUBST+=	EMACS_INFOPREFIX=${EMACS_INFOPREFIX:C|^${PREFIX}/||}
-_EMACS_PLIST_SUBST+=	EMACS_LISPPREFIX=${EMACS_LISPPREFIX:C|^${PREFIX}/||}
-
-_EMACS_PLIST_SUBST+=	FOR_emacs_no_byte_compile="${${EMACS_VERSION_MAJOR}>22:?@comment :}"
-
-PLIST_SUBST+=		${_EMACS_PLIST_SUBST}
-
-PRINT_PLIST_AWK+=	{ gsub(/${EMACS_LISPPREFIX:S|${PREFIX}/||:S|/|\\/|g}/, \
-			"$${EMACS_LISPPREFIX}"); \
-			print; next; }
-
-#
-# Build environment (buildlink3)
-#
-
-.if defined(EMACS_BUILDLINK)
-_EMACS_DIR=	${BUILDLINK_DIR}/share/emacs
-.  if ${EMACS_TYPE} == emacs27
-ALL_ENV+=	EMACSLOADPATH=${_EMACS_DIR}/${_EMACS_VERSION_MAJOR}.${_EMACS_VERSION_MINOR}.${_EMACS_VERSION_MICRO}/lisp:${_EMACS_DIR}/site-lisp
-.  else
-ALL_ENV+=	EMACSLOADPATH=${_EMACS_DIR}/${_EMACS_VERSION_MAJOR}.${_EMACS_VERSION_MINOR}/lisp:${_EMACS_DIR}/site-lisp
-.  endif
-.include	"${_EMACS_PKGDIR}/buildlink3.mk"
-.endif
-
-.endif	# EMACS_MK
+.endif	# EMACS_GIT_MODULES_MK
diff --git a/emacs-git/options.mk b/emacs-git/options.mk
index d4f4345a37..49d0478fb9 100644
--- a/emacs-git/options.mk
+++ b/emacs-git/options.mk
@@ -2,8 +2,8 @@
 
 ### Set options
 PKG_OPTIONS_VAR=			PKG_OPTIONS.emacs
-PKG_SUPPORTED_OPTIONS=			dbus gconf gnutls imagemagick jansson svg xaw3d xft2 xml
-PKG_SUGGESTED_OPTIONS=			jansson
+PKG_SUPPORTED_OPTIONS=			dbus gconf gnutls imagemagick jansson svg xaw3d xft2 xml debug
+PKG_SUGGESTED_OPTIONS=			jansson debug
 # xaw3d is only valid with tookit = xaw
 
 PKG_OPTIONS_OPTIONAL_GROUPS+=		window-system
@@ -25,6 +25,11 @@ PKG_SUGGESTED_OPTIONS=	dbus gconf gnutls gtk3 svg xaw3d xft2 xml x11
 
 .include "../../mk/bsd.options.mk"
 
+.if !empty(PKG_OPTIONS:Mdebug)
+# For wip special, following gives symbol names when back trace is shown.
+CFLAGS+=		-g
+.endif
+
 ### ---	 Check window-system independent options first
 ###
 ### Support D-BUS
@@ -130,6 +135,8 @@ CONFIGURE_ARGS+=	--without-xft --without-libotf --without-m17n-flt
 USE_TOOLS+=		pkg-config
 .include "../../x11/gtk3/buildlink3.mk"
 CONFIGURE_ARGS+=	--with-x-toolkit=gtk3
+.include "../../graphics/cairo/buildlink3.mk"
+CONFIGURE_ARGS+=	--with-cairo
 .  elif !empty(PKG_OPTIONS:Mgtk2) || !empty(PKG_OPTIONS:Mgtk)
 USE_TOOLS+=		pkg-config
 .include "../../x11/gtk2/buildlink3.mk"
diff --git a/emacs-nox11-git/Makefile b/emacs-nox11-git/Makefile
index fe80e9b761..2093d4b98a 100644
--- a/emacs-nox11-git/Makefile
+++ b/emacs-nox11-git/Makefile
@@ -1,43 +1,39 @@
-# $NetBSD: Makefile,v 1.8 2012/11/23 22:33:29 othyro Exp $
+# $NetBSD$
 
-DISTNAME=	emacs-nox11-24.0.90
-PKGREVISION=	${_GIT_PKGVERSION:S/.//g}
-CATEGORIES=	editors
-GIT_REPO=	git://repo.or.cz/emacs.git
-
-MAINTAINER=	pkgsrc-users@NetBSD.org
-HOMEPAGE=	http://www.gnu.org/software/emacs/emacs.html
-COMMENT=	GNU editing macros (editor, no x11)
-LICENSE=	gnu-gpl-v3
-
-USE_TOOLS+=		gmake makeinfo gzip autoconf automake
-# for example, Mac OS X 10.6.1 has 2.61 which fails
-AUTOCONF_REQUIRED=	2.65
-
-CONFLICTS=	emacs-[0-9]*
-
-BUILD_TARGET=	bootstrap
+.include "../../wip/emacs-git/Makefile.common"
+.include "version.mk"
+.include "options.mk"
 
-GNU_CONFIGURE=		yes
-INFO_FILES=		yes
-REPLACE_PERL=		lib-src/grep-changelog
-CHECK_INTERPRETER_SKIP=	bin/grep-changelog
+DISTNAME:=	${DISTNAME:S/-/-nox11-/}
+PKGNAME:=	${PKGNAME:S/-/-nox11-/}
+PKGREVISION=	${_GIT_PKGVERSION:S/.//g}
 
 DISTINFO_FILE=	${PKGDIR}/../emacs-git/distinfo
 PATCHDIR=	${PKGDIR}/../emacs-git/patches
 DESCR_SRC=	${PKGDIR}/../emacs-git/DESCR
-#ESSAGE_SRC=	${PKGDIR}/../emacs-git/MESSAGE
+#MESSAGE_SRC=	${PKGDIR}/../emacs-git/MESSAGE
 PLIST_SRC=	${PKGDIR}/../emacs-git/PLIST
 
-WRKSRC=		${WRKDIR}/emacs
-
-CONFIGURE_ARGS+=	--without-x	# nox11
-CONFIGURE_ARGS+=	--srcdir=${WRKSRC}
-CONFIGURE_ARGS+=	--localstatedir=${VARBASE}
-CONFIGURE_ENV+=		GAMEOWN=${GAMEOWN:Q}
+COMMENT=	GNU editing macros (editor, no x11)
 
-BUILD_DEFS+=		VARBASE
-BUILD_DEFS+=		GAMEDATAMODE
+CONFLICTS=	emacs${_EMACS_VERSION_MAJOR}-nox11-[0-9]*
+
+CONFIGURE_ARGS+=	--without-dbus
+CONFIGURE_ARGS+=	--without-xml2
+CONFIGURE_ARGS+=	--without-gnutls
+CONFIGURE_ARGS+=	--without-rsvg
+CONFIGURE_ARGS+=	--without-gconf
+CONFIGURE_ARGS+=	--without-imagemagick
+CONFIGURE_ARGS+=	--without-xaw3d
+CONFIGURE_ARGS+=	--without-xft --without-libotf --without-m17n-flt
+CONFIGURE_ARGS+=	--without-ns
+
+CONFIGURE_ARGS+=	--without-x
+CONFIGURE_ARGS+=	--without-xpm
+CONFIGURE_ARGS+=	--without-jpeg
+CONFIGURE_ARGS+=	--without-tiff
+CONFIGURE_ARGS+=	--without-gif
+CONFIGURE_ARGS+=	--without-png
 
 #SUBST_CLASSES+=		test_equal
 #SUBST_STAGE.test_equal=	pre-configure
@@ -45,22 +41,9 @@ BUILD_DEFS+=		GAMEDATAMODE
 #SUBST_FILES.test_equal=	mac/make-package
 #SUBST_SED.test_equal=		-e 's/ == / = /g'
 
-MAKE_DIRS_PERMS+=	${VARBASE}/games/emacs ${GAMEOWN:Q} ${GAMEGRP} ${GAMEDIRMODE:Q}
-CONF_FILES_PERMS+=	/dev/null ${VARBASE}/games/emacs/snake-scores ${GAMEOWN:Q} ${GAMEGRP:Q} ${GAMEDATAMODE}
-CONF_FILES_PERMS+=	/dev/null ${VARBASE}/games/emacs/tetris-scores ${GAMEOWN:Q} ${GAMEGRP:Q} ${GAMEDATAMODE}
-
-# build PATH in the dumped emacs is not a problem
-CHECK_WRKREF_SKIP+=	bin/emacs
-CHECK_WRKREF_SKIP+=	bin/emacs-${PKGVERSION_NOREV}
-
-FILESDIR=	${.CURDIR}/../../editors/emacs/files
-
-post-extract:
-	${CP} ${FILESDIR}/site-init.el ${WRKSRC}/lisp
+FILESDIR=	${.CURDIR}/../../wip/emacs-git/files
 
-pre-configure:
-	(cd ${WRKSRC}; ./autogen.sh)
+## Share _GIT_DISTFILE.default with wip/emacs-git
+_GIT_DISTFILE.default=${_EMACS_FLAVOR}${_EMACS_VERSION_MAJOR}-default-gitarchive.tar.gz
 
-.include "../../wip/mk/git-package.mk"
-.include "../../mk/oss.buildlink3.mk"
 .include "../../mk/bsd.pkg.mk"
diff --git a/emacs-nox11-git/options.mk b/emacs-nox11-git/options.mk
new file mode 100644
index 0000000000..c06bf7a537
--- /dev/null
+++ b/emacs-nox11-git/options.mk
@@ -0,0 +1,27 @@
+# $NetBSD$
+
+### Set options
+PKG_OPTIONS_VAR=			PKG_OPTIONS.emacs
+PKG_SUPPORTED_OPTIONS=			gnutls jansson
+PKG_SUGGESTED_OPTIONS=			gnutls jansson
+
+.include "../../mk/bsd.options.mk"
+
+###
+### Support gnutls
+###
+.if !empty(PKG_OPTIONS:Mgnutls)
+.include "../../security/gnutls/buildlink3.mk"
+.include "../../security/p11-kit/buildlink3.mk"
+.else
+CONFIGURE_ARGS+=	--without-gnutls
+.endif
+
+###
+### Support jansson (JSON library)
+###
+.  if !empty(PKG_OPTIONS:Mjansson)
+.include "../../textproc/jansson/buildlink3.mk"
+.  else
+CONFIGURE_ARGS+=	--without-json
+.  endif
diff --git a/emacs-nox11-git/version.mk b/emacs-nox11-git/version.mk
new file mode 100644
index 0000000000..eb8639a892
--- /dev/null
+++ b/emacs-nox11-git/version.mk
@@ -0,0 +1,5 @@
+# $NetBSD$
+
+.include "../../wip/emacs-git/version.mk"
+
+_EMACS_REQD=	emacs27-nox11>=27
